### ISSUE TITLE
Export problem

### ISSUE THREAD

**Issue by linqFR** [unjs/defu#27]
@pi0 : importing into a 'module' type app (in package.json, type:'module')  triggers an error

```
node:internal/process/esm_loader:74
    internalBinding('errors').triggerUncaughtException(
                              ^

Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: No "exports" main defined in ...\node_modules\defu\package.json imported from app.js
    at new NodeError (node:internal/errors:363:5)
    at throwExportsNotFound (node:internal/modules/esm/resolve:321:9)
    at packageExportsResolve (node:internal/modules/esm/resolve:510:7)
    at packageResolve (node:internal/modules/esm/resolve:687:14)
    at moduleResolve (node:internal/modules/esm/resolve:739:18)
    at Loader.defaultResolve [as _resolve] (node:internal/modules/esm/resolve:853:11)
    at Loader.resolve (node:internal/modules/esm/loader:89:40)
    at Loader.getModuleJob (node:internal/modules/esm/loader:242:28)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:73:40)
    at link (node:internal/modules/esm/module_job:72:36) {
  code: 'ERR_PACKAGE_PATH_NOT_EXPORTED'
}
```

**Comment by linqFR** [comment]
i think the problem comes from your package.json

https://nodejs.org/api/packages.html#packages_conditional_exports

**Comment by pi0** [comment]
Ohh thanks for the notice! Should be fixed in 4.0.1 :)
